### PR TITLE
feat: add missing pattern context to demo and cover render

### PR DESCRIPTION
**Pull request by noyoucmon**
This adds pattern missing context to demo and cover render. It also awaits the render for cover.

### CODE REVIEW

**Comment by noyoucmon** [comment]
This also relates to https://github.com/patternplate/patternplate/issues/213.

File: packages/cli/src/build/index.js [hunk 1]
```diff
@@ -100,10 +101,10 @@ async function build({flags}) {
   // Create demo.html files
   await Promise.all(patterns.map(async pattern => {
     const component = getComponent(bundles, pattern);
-    const context = getContext(pattern);
+    const context = getPatternContext(pattern);
```
Comment (marionebl) [inline packages/cli/src/build/index.js]: Would be great if we could await those, too.

File: packages/cli/src/build/index.js [hunk 2]
```diff
@@ -88,9 +88,10 @@ async function build({flags}) {
   // Create /
   if (typeof cover === "string") {
     const cover = getModule(COVER_PATH);
+    const context = getCoverContext(config);
     const result = typeof cover.render === "function"
-      ? cover.render(cover)
-      : getModule(RENDER_PATH)(cover);
+      ? cover.render(cover, context)
```
Comment (marionebl) [inline packages/cli/src/build/index.js]: Would be great if we could await those, too.

Comment (marionebl) [reply packages/cli/src/build/index.js]: I thinks binding the relevant function to a `render` is a good idea in all cases, that should help with readability when applying the `Promise.resolve` guard for async execution

File: packages/cli/src/build/index.js (hunk 1 above)
Comment (noyoucmon) [reply packages/cli/src/build/index.js]: Changed in new commit.

File: packages/cli/src/build/index.js (hunk 2 above)
Comment (noyoucmon) [reply packages/cli/src/build/index.js]: Changed in new commit.

**Comment by marionebl** [comment]
Thanks!

**Pull request merged**